### ISSUE TITLE
st_cast not able to handle mutlicurve to multilinestring anymore (unknown WKB type 11)? similar issue to #1194 from 2019

### ISSUE THREAD

**Issue by bpfree** [r-spatial/sf#2203]
I seem to be running into a similar problem as this individual did back in 2019 (#1194). This is strange as the below code worked for me weeks/months ago, but now am getting the similar error:

```
Error in scan(text = lst[[length(lst)]], quiet = TRUE) : 
  scan() expected 'a real', got 'ParseException:'
Error in (function (msg)  : ParseException: Unknown WKB type 11
```

Source for the data came from [here](https://marinecadastre.gov/downloads/data/mc/SubmarineCable.zip).

```
submarine_cables_noaa <- sf::st_read(dsn = submarine_cable_dir,
                                     layer = paste(sf::st_layers(dsn = submarine_cable_dir,
                                                                 do_count = TRUE))) %>%
  # change to multilinestring (for 1 feature is multicurve: 1171)
  sf::st_cast(to = "MULTILINESTRING") %>%
  # make sure all geometries are valid
  sf::st_make_valid() %>%
  # reproject data into a coordinate system (NAD 1983 UTM Zone 10N) that will convert units from degrees to meters
  sf::st_transform("EPSG:26910") # EPSG 26910 (https://epsg.io/26910)
```

Any ideas what may have caused sf to no longer be able to cast multicurve to multilinestrings? Not sure if the [update for version 1.0-13](https://r-spatial.github.io/sf/news/index.html) was what affected it. Though, I have tried to use older versions (e.g., 1.0-9 and 1.0-12), yet have gotten similar errors, so maybe it suggests something is wrong on my end?

`sessionInfo`
```
R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS 13.4.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tidyr_1.3.0         terra_1.7-40        stringr_1.5.0       sf_1.0-14           rnaturalearth_0.3.3 rmapshaper_0.5.0    rgeos_0.6-3        
 [8] rgeoda_0.0.10-4     digest_0.6.31       rgdal_1.6-7         plyr_1.8.8          pdftools_3.3.3      ncf_1.3-2           janitor_2.2.0      
[15] ggplot2_3.4.2       fs_1.6.2            fasterize_1.0.4     elsa_1.1-28         raster_3.6-23       sp_2.0-0            dplyr_1.1.2        
[22] docxtractr_0.6.5   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.11         lubridate_1.9.2     lattice_0.21-8      png_0.1-8           class_7.3-22        utf8_1.2.3          V8_4.3.2           
 [8] R6_2.5.1            cellranger_1.1.0    e1071_1.7-13        httr_1.4.6          pillar_1.9.0        tabulizer_0.2.3     rlang_1.1.1        
[15] curl_5.0.1          readxl_1.4.3        rstudioapi_0.14     qpdf_1.3.2          tabulizerjars_1.0.1 munsell_0.5.0       proxy_0.4-27       
[22] compiler_4.2.0      askpass_1.1         pkgconfig_2.0.3     tidyselect_1.2.0    tibble_3.2.1        codetools_0.2-19    fansi_1.0.4        
[29] withr_2.5.0         rhdf5filters_1.10.1 grid_4.2.0          jsonlite_1.8.4      lwgeom_0.2-11       gtable_0.3.3        lifecycle_1.0.3    
[36] DBI_1.1.3           pacman_0.5.1        magrittr_2.0.3      units_0.8-2         scales_1.2.1        KernSmooth_2.23-20  cli_3.6.1          
[43] stringi_1.7.12      remotes_2.4.2       snakecase_0.11.0    xml2_1.3.3          generics_0.1.3      vctrs_0.6.3         Rhdf5lib_1.20.0    
[50] tools_4.2.0         glue_1.6.2          purrr_1.0.1         timechange_0.2.0    colorspace_2.1-0    rhdf5_2.42.1        classInt_0.4-9     
[57] rJava_1.0-6
```

**Comment by edzer** [comment]
This is collateral damage from the fix for #1961: MultiCurve is not supported by GEOS, and breaks the `st_is_empty()` check. You could, for the time being, pick out the multicurve and only cast that:
```r
> class(st_geometry(x[1171,]))
[1] "sfc_MULTICURVE" "sfc"           
> x$Shape[1171] = st_cast(x$Shape[1171], "MULTILINESTRING")
> x
Simple feature collection with 2316 features and 4 fields
Geometry type: MULTILINESTRING
Dimension:     XY
Bounding box:  xmin: -180 ymin: 4.999994 xmax: 180 ymax: 61.19428
Geodetic CRS:  NAD83
First 10 features:
...
```

**Comment by bpfree** [comment]
Thanks -- good to know that it was a change from a previous update. Yeah, that is what I ended up doing to alleviate the situtation.
